src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -131,6 +134,8 @@ public void Reset()
             {
                 httpConnectionFeature.IsLocal = false;
             }
+
+            _currentITlsConnectionFeature = _tlsConnectionFeature;
```

Comment:
Instead of changing ConnectoinFilterContext, I think it would be a better to add a method to IConnectionFilter (maybe PrepareRequest?) that takes an IFeatureCollection as a parameter. We could use this same PrepareRequest method to [properly set the scheme](https://github.com/aspnet/KestrelHttpServer/issues/365).